Assets/_PlatformerDevelopment/Scripts/Managers/StateManager.cs

```diff
@@ -0,0 +1,50 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="StateManager.cs" company="PERIGON GAMES">
+//     Copyright (c) PERIGON GAMES. 2020 All Rights Reserved
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+
+namespace PersonalDevelopment
+{
+    public enum State
+    {
+        StartMenu,
+        CharacterSelection,
+        PreGame,
+        Play,
+        EndGame
```

Comment:
I still feel that it would be better, generally, to have a more explicit statemachine (like Unity has for switching between [Play, and Pause](https://docs.unity3d.com/ScriptReference/PlayModeStateChange.html)) where we would have Entering-State and Exiting State. This will probably only be needed for like Play, Pregame, and character selection

Reply:
This would help so that we can handle initializing event/stuff on the Starting-State regardless of which state we came from and handle the destroy/ uninitializing stuff event/stuff on the Exiting-state regardless of which state we are going to next

Reply:
@Gursaanj that's a good idea. But do you think it's better to have generic ones now? or just add them as we go along and put more into design?

Reply:
@m4rtint  hmm good question. Since I guess we don't know the exact states now, we can add them after we confirm the design stuff!. But in general, I think Entering-Play, and Exiting-Play will be generic enough?? We can always discuss and finalize this at another time